libevmasm/SemanticInformation.cpp

```diff
@@ -29,6 +29,128 @@ using namespace std;
 using namespace solidity;
 using namespace solidity::evmasm;
 
+vector<SemanticInformation::Operation> SemanticInformation::readWriteOperations(Instruction _instruction)
+{
+	switch (_instruction)
+	{
+	case Instruction::SSTORE:
+	case Instruction::SLOAD:
+	case Instruction::MSTORE:
+	case Instruction::MSTORE8:
+	case Instruction::MLOAD:
+	case Instruction::REVERT:
+	case Instruction::RETURN:
+	case Instruction::EXTCODECOPY:
+	case Instruction::CODECOPY:
+	case Instruction::CALLDATACOPY:
+	case Instruction::RETURNDATACOPY:
+	case Instruction::KECCAK256:
+	case Instruction::LOG0:
+	case Instruction::LOG1:
+	case Instruction::LOG2:
+	case Instruction::LOG3:
+	case Instruction::LOG4:
+	{
+		Operation op;
+		if (memory(_instruction) == Effect::Write || storage(_instruction) == Effect::Write)
+			op.effect = Effect::Write;
+		else
+			op.effect = Effect::Read;
```

Comment:
Maybe an assert here that `memory(_instruction) != Effect::None` and the same for storage?